pandas/tests/test_multilevel.py

```diff
@@ -290,6 +290,22 @@ def test_datetime_object_multiindex(self):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_multiindex_with_na(self):
+        df = DataFrame(
+            [
+                ["A", np.nan, 1.23, 4.56],
+                ["A", "G", 1.23, 4.56],
+                ["A", "D", 9.87, 10.54],
+            ],
+            columns=["pivot_0", "pivot_1", "col_1", "col_2"],
+        )
+        df.set_index(["pivot_0", "pivot_1"], inplace=True)
```

Comment:
```suggestion
        ).set_index(["pivot_0", "pivot_1"])
```

Reply:
May I ask what benefit do we gain from chaining the operations?

Reply:
We want to avoid `inplace=True` where possible